src/Interpreters/ExecuteScalarSubqueriesVisitor.cpp

```diff
@@ -204,7 +204,13 @@ void ExecuteScalarSubqueriesMatcher::visit(const ASTSubquery & subquery, ASTPtr
 
                 ast_new->setAlias(ast->tryGetAlias());
                 ast = std::move(ast_new);
-                return;
+
+                /// Empty subquery result is equivalent to NULL
```

Comment:
Why `NULL`? You use a default value for the corresponding type

Reply:
It's a current behavior. The problem was that in such case we didn't save block with NULL into `scalars` and it affected query plan we had built.

```sql
:) SELECT (SELECT 1 WHERE 0) as x;

┌────x─┐
│ ᴺᵁᴸᴸ │
└──────┘
```